hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -417,4 +425,78 @@ private List<ContainerBlockMetadata> getBlocks(
   private BucketLayout getBucketLayout() {
     return BucketLayout.DEFAULT;
   }
+
+  @GET
+  @Path("insights/containermismatch")
+  public Response getContainerMisMatchInsights() {
+    List<ContainerDiscrepancyInfo> containerDiscrepancyInfoList =
+        new ArrayList<>();
+    try {
+      Map<Long, ContainerMetadata> omContainers =
+          reconContainerMetadataManager.getContainers(-1, -1);
+      List<Long> scmContainers = containerManager.getContainers().stream()
```

Comment:
Assume this scmContainers are which are is not in DELETED state?

Reply:
Yes, this is SCM container list irrespective of their state.

Reply:
1-In this case all SCM Containers when compared with OM Containers , will show all Deleted State containers as discrepancy i.e missing in OM
2-Also OM will never report any container "available in OM but missing in SCM" as deleted container is  still in SCM list.
We need to have only Non-Deleted State SCM containers.
@sumitagrawl can you help check

Reply:
@devmadhuu @krishnaasawa1 Yes, we need consider non-deleted state of containers. This is like a dataloss only if OM have referred container but that container is not present / deleted.

Reply:
> OM Containers
@sumitagrawl @krishnaasawa1  first point is fixed where "Missing in OM and present in SCM" case.